### ISSUE TITLE
Parallel Requests Handling in Leshan Server

### ISSUE THREAD

**Issue by allaismail** [eclipse-leshan/leshan#1599]
### Question

Hello everyone,

I have a question regarding the request handling structure of the Demo server. After conducting some tests, I've observed that the server seems unable to handle multiple requests in parallel. Instead, it queues the requests, waiting for the first request to be completed before processing subsequent ones.

For instance, when I send a file that takes 3 minutes to transmit, any other actions I attempt to perform on the server are delayed until the transmission completes.

I would appreciate any insights or suggestions on how to enable parallel request handling in the server to improve its responsiveness and efficiency.

Thank you.

Best regards,

**Comment by sbernard31** [comment]
> After conducting some tests, I've observed that the server seems unable to handle multiple requests in parallel. Instead, it queues the requests, waiting for the first request to be completed before processing subsequent ones.

Reading the code, and testing it quickly I didn't think this is true. At least I was able to send 2 requests in parallel, blocking both answer. Wait for a moment. Unblock the second one (server get the answer), then Unblock the first one. (server get the answer)

> For instance, when I send a file that takes 3 minutes to transmit, any other actions I attempt to perform on the server are delayed until the transmission completes.

I tested with block transfer too and I didn't see that behavior.

> I would appreciate any insights or suggestions on how to enable parallel request handling in the server to improve its responsiveness and efficiency.

I don't know exactly what you do with server demo but I'm a bit surprised that you have this kind of requirement just for a demo... :thinking:

**Comment by sbernard31** [comment]
Oh and funny to `Thumbs up` :+1:  its own issue :grin

**Comment by allaismail** [comment]
Hello sbernard,

I apologize for the confusion. You were correct. Upon retesting, I successfully connected three clients using Wakaama to the server demo. Additionally, I sent a one-megabyte file as a firmware package to these three clients simultaneously without encountering any issues.

thank you 

you can close the issue

**Comment by sbernard31** [comment]
You're welcome.
